splice_machine/src/main/java/com/splicemachine/derby/stream/iterator/GetNLJoinIterator.java

```diff
@@ -57,53 +54,47 @@ protected OperationContext getCtx() {
         return ctx;
     }
 
-    private boolean needsLCCInContext() {
+    private void marshallTransactionForTrigger() throws Exception{
         OperationContext ctx = getCtx();
         if (ctx != null) {
             Activation activation = ctx.getActivation();
             if (activation != null) {
                 LanguageConnectionContext lcc = activation.getLanguageConnectionContext();
                 if (lcc != null) {
                     TriggerExecutionContext tec = lcc.getTriggerExecutionContext();
-                    if (tec != null)
-                        return tec.currentTriggerHasReferencingClause();
+                    if (tec != null && tec.currentTriggerHasReferencingClause()) {
+                            ah = new ActivationHolder(activation, null);
+                            LanguageConnectionContext newLCC = null;
+                            try {
+                                LanguageConnectionContext oldLCC;
+                                oldLCC = lcc;
+                                ah.newTxnResource();
```

Comment:
I still have reservations about this code. A new ContextManager is created underneath with a new LCC. I see that there are leftover ContextManagers stuck in ThreadLocal of Spark executor threads (not necessarily from this code). That's one step (if not there already) from a fully fledged memory leak.
The question remains, why is it needed? Should it be addressed at a higher level? Don't nested ContextManagers in SpliceTransactionManager help?